docs/plugin_developers_guide/10_testing.rst

```diff
@@ -466,3 +466,429 @@ You can also check your default configuration. Although in Python this is usuall
+       }
+   }
+
+**Validation Behavior:**
+   - Parses ``opc.tcp://`` URL format
+   - Tests host reachability to ``localhost``
+   - Tests service listening on ``localhost:4840``
+
+**Supported URL Schemes:**
+   - ``http://``, ``https://``
+   - ``opc.tcp://`` (OPC UA)
+   - ``tcp://`` (MQTT)
+   - ``mqtt://``, ``mqtts://``
+   - ``ftp://``
+
+Server and Port Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Generic server configuration pattern:
+
+.. code-block:: json
+
+   {
+       "plugin": {
+           "description": "Database North Plugin",
+           "type": "string",
+           "default": "database",
+           "readonly": "true"
+       },
+       "ServerHostname": {
+           "description": "Database server hostname",
+           "type": "string",
+           "default": "db.example.com",
+           "order": "1",
+           "displayName": "Server Hostname"
+       },
+       "ServerPort": {
+           "description": "Database server port",
+           "type": "integer",
+           "default": "5432",
+           "order": "2",
+           "displayName": "Server Port"
+       }
+   }
+
+**Validation Behavior:**
+   - Tests host reachability to ``db.example.com``
+   - Tests service listening on ``db.example.com:5432``
+
+Default Port Assignment
+~~~~~~~~~~~~~~~~~~~~~~~
+
+When no explicit port is configured, the validation system applies default ports based on field names and common protocols:
+
+**IP Fields** (``ip``, ``IP``)
+   - ``102`` (S7 PLC)
+   - ``44818`` (EtherNet/IP)
+   - ``502`` (Modbus TCP)
+   - ``80`` (HTTP)
+   - ``443`` (HTTPS)
+
+**Address Fields** (``address``)
+   - ``502`` (Modbus TCP)
+   - ``80`` (HTTP)
+   - ``443`` (HTTPS)
+
+**Host/Server Fields** (``host``, ``hostname``, ``server``)
+   - ``80`` (HTTP)
+   - ``443`` (HTTPS)
+
+**Broker Fields** (``broker``, ``brokerhost``)
+   - ``1883`` (MQTT)
+   - ``8883`` (MQTTS)
+
+Testing Configuration Validation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**API Endpoint:** ``PUT /fledge/plugin/validate``
+
+**Request Body:** JSON configuration object
+
+**Example Test Request:**
+
+.. code-block:: bash
+
+   curl -X PUT http://localhost:8081/fledge/plugin/validate \
+        -H "Content-Type: application/json" \
+        -d '{
+            "plugin": {
+                "description": "Test Plugin",
+                "type": "string",
+                "default": "test",
+                "readonly": "true"
+            },
+            "address": {
+                "description": "Server address",
+                "type": "string",
+                "default": "192.168.1.100"
+            },
+            "port": {
+                "description": "Server port",
+                "type": "integer",
+                "default": "502"
+            }
+        }'
+
+Validation Response Format
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The validation API returns a JSON response with test results:
+
+**Success Response (HTTP 200):**
+
+.. code-block:: json
+
+   {
+       "HostReachable": {
```

Comment:
Please avoid mixing cases. What is the convention we are following throughout - camelCase, snake_case, what? 

We are doing a poor job in writing semantically incorrect endpoints URLs and request / response structure with giving proper thoughts. And this is happeing every now and then for many services.

Reply:
We need to establish a standard approach to prevent mixing these conventions across.

Also, For configuration items in plugins, we mostly use snake_case in Python plugins, whereas in C, we often use camelCase or a combination of both.

Reply:
These names date from when we adopted no description in the spec and can be camelCase now that we don;t want to use them in messages.